File: ros2-basic/topic_simple/src/node_helloworld_py/node_helloworld_py/node_subscriber_by_std_msgs.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class NodeHelloworld(Node):
    def __init__(self, name: str):
        super().__init__(name)
        self.subscription_ = self.create_subscription(
            String, "helloworld_py_by_std_msgs", self.subscriber_callback, 10
        )

    def subscriber_callback(self, msg):
        self.get_logger().info(msg.data)


def main(args=None):
    rclpy.init(args=args)
    node = NodeHelloworld("node_topic_subscriber_helloworld_py_by_std_msgs")
    rclpy.spin(node)
    rclpy.shutdown()

File: ros2-basic/topic_simple/src/node_helloworld_cpp/src/node_subscriber_by_std_msgs.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class NodeHelloworld : public rclcpp::Node
{
public:
    NodeHelloworld(std::string name) : Node(name)
    {
        subscription_ = this->create_subscription<std_msgs::msg::String>(
            "helloworld_cpp_by_std_msgs",
            rclcpp::QoS(rclcpp::KeepLast(10)),
            std::bind(&NodeHelloworld::subscriber_callback, this, std::placeholders::_1));
    }

private:
    void subscriber_callback(const std_msgs::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "I heard: '%s'", msg->data.c_str());
    }

private:
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NodeHelloworld>("node_topic_subscriber_helloworld_cpp_by_std_msgs");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}